renderdoc/driver/vulkan/vk_linux.cpp

```diff
@@ -246,9 +246,9 @@ string LayerRegistrationPath(LayerPath path)
       const char *xdg = getenv("XDG_DATA_HOME");
       if(xdg && FileIO::exists(xdg))
         return string(xdg) + "/vulkan/implicit_layer.d/renderdoc_capture.json";
-
-      return string(getenv("HOME")) +
-             "/.local/share/vulkan/implicit_layer.d/renderdoc_capture.json";
+      const char *home = getenv("HOME");
+      if(home != NULL)
+        return string(home) + "/.local/share/vulkan/implicit_layer.d/renderdoc_capture.json";
```

Comment:
In the code below, in case $HOME is empty the path will be in a renderdoc owned folder .renderdoc that will be created by RenderDoc.
In this case, the path will point to /.local/share/vulkan/implicit_layer.d/renderdoc_capture.json" which may/may not exit, or may/may not have access rights. Are you certain you want the code to assume that location?..

Reply:
Even under `/home` the folder may not exist or have access rights. No changes are made without the user explicitly requesting them, and they'll see the path listed without `/home` when prompted about it. I'm more concerned that there's some code assuming some kind of path that will break with an empty string.